### PR TITLE
Port ridge regression to JavaScript

### PR DESCRIPTION
**Pull request by swashcap**
Move the computation’s core functionality from Python to JavaScript.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,10 +11,11 @@
   ],
   "description": "Single shot implementation of ridge regression in python for COINSTAC.",
   "dependencies": {
-    "freesurfer-parser": "0.0.2"
+    "freesurfer-parser": "0.0.2",
+    "numeric": "^1.2.6"
```
Comment (swashcap) [inline package.json:15]: `numeric` must be added to _package.json_ such that other developers and COINSTAC itself has the dependency. This is a Node.js convention.

File: package.json [hunk 2]
```diff
@@ -11,10 +11,11 @@
...
   },
   "devDependencies": {
-    "coinstac-simulator": "^2.1.1",
+    "coinstac-simulator": "^2.2.0",
```
Comment (swashcap) [inline package.json:18]: MRN-Code/coinstac@f2b3d87908305f4422b10f4d61e86a36c5d8369f added some performance improvements. You’ll want to ensure you have the updated dependency by running:

``` shell
npm prune && npm install
```

File: src/ridge_regress.py [hunk 3]
```diff
@@ -1,55 +0,0 @@
-
```
Comment (swashcap) [inline src/ridge_regress.py:1]: We don’t need this file. It’s in git if we want to pull anything back. :wave:

File: src/index.js [hunk 4]
```diff
@@ -98,24 +99,36 @@ module.exports = {
...
+    },
   }],
   remote: {
     type: 'function',
     fn(opts) {
       const userResults = opts.userResults;
 
       // Not all user results contain betas. Return early.
-      if (userResults.some(userResult => !((userResult || {}).data || {}).beta_vector)) {
+      if (userResults.some(userResult => !((userResult || {}).data || {}).betaVector)) {
```
Comment (swashcap) [inline src/index.js:125]: Out style guide prefers camel case to snake case.

File: src/index.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const regression=require('./regression');
+const regression = require('./regression');
```
Comment (swashcap) [inline src/index.js:3]: We use AirBnB’s style guide to write consistent code. Check for errors by running:

``` shell
npm run lint
```

This is also wired up to git’s pre-commit hook.

File: src/index.js [hunk 6]
```diff
@@ -99,31 +99,24 @@ module.exports = {
       type: 'covariates',
     }],
   }, {
-// oneshotridgeregression begin 
     type: 'function',
-//    cmd: 'python',
```
Comment (swashcap) [inline src/index.js]: Don’t need this old stuff. It’s in history if we need to reference.

File: src/index.js [hunk 7]
```diff
@@ -99,31 +99,24 @@ module.exports = {
...
-//    args: ['./ridge_regress.py'],
-//    verbose: true,
-   // get dimension of the data
     fn(opts) {
+      const previousData = opts.previousData;
 
-   //   const beta=regression.oneShot(opts.previousData.X,opts.previousData.y)
-   //    console.log('beta vector is:',beta);
-   //   return beta 
-    const previousData=opts.previousData;
```
Comment (swashcap) [inline src/index.js]: Diff is ugly here. These are mostly whitespace changes to appease the linter.

File: package.json (hunk 1 above)
Comment (JingMing623) [reply package.json:15]: Okay, will remember to include the dependency into package.json next time

Comment (JingMing623) [inline package.json]: Okay, will remember to include the dependency into package.json next time.

File: package.json (hunk 2 above)
Comment (JingMing623) [reply package.json:18]: Okay, so we released new coinstac-simulator.

**Comment by swashcap** [comment]
Mergin’ this. We can make modifications on another feature branch if needed. 🎉

**Comment by JingMing623** [comment]
Great!

**Pull request merged**